splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java

```diff
@@ -1404,6 +1406,66 @@ public boolean useTxnAwareCache() {
         return !SpliceClient.isRegionServer;
     }
 
+    @Override
+    public long getConglomerateCreationTxId(long tableConglom) throws StandardException {
+        try {
+            SIDriver driver = SIDriver.driver();
+            if(SanityManager.DEBUG) {
+                SanityManager.ASSERT(driver != null);
+            }
+            PartitionFactory tableFactory = driver.getTableFactory();
+            if(SanityManager.DEBUG) {
+                SanityManager.ASSERT(tableFactory != null);
+            }
+            Partition partition = tableFactory.getTable(Long.toString(tableConglom));
+            if(SanityManager.DEBUG) {
+                SanityManager.ASSERT(partition != null);
+            }
+            PartitionAdmin admin = tableFactory.getAdmin();
+            if(SanityManager.DEBUG) {
+                SanityManager.ASSERT(admin != null);
+            }
+            com.splicemachine.access.api.TableDescriptor descriptor = admin.getTableDescriptor(Utils.constructHbaseName(partition.getTableName()));
+            if(SanityManager.DEBUG) {
+                SanityManager.ASSERT(descriptor != null);
+            }
+            return Long.parseLong(descriptor.getTransactionId());
```

Comment:
There's old tables/indexes that don't have a transaction id, this would fail for them and make them inaccessible. I think we can assume they are always older than the requested TXN

Reply:
Thanks, I didn't know that, I fixed it assuming we get `null` from `TableDescriptor.getTransactionId` for these old indexes / tables, and return 0 instead allowing the optimizer to consider them for time-travel queries.